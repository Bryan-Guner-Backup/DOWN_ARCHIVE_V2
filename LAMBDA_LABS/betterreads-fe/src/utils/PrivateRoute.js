import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { user } from '../utils/helpers';

export const PrivateRoute = ({ component: Component, ...rest }) => {
  return (
    <Route
      {...rest}
      render={(props) => {
        if (localStorage.getItem('token') !== null) {
          return <Component {...props} />;
        } else {
          return <Redirect to='/signin' />;
        }
      }}
    />
  );
};
