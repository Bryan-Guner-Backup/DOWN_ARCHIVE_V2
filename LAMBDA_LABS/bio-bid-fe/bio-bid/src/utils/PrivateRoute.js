import React from "react";
import { Route, Redirect } from "react-router-dom";
import { useOktaAuth } from "@okta/okta-react";

const PrivateRoute = ({ component: Component, ...rest }) => {
  const { authState } = useOktaAuth();

  return (
    // Denies users from manually typing in url's to reach dashboard
    <Route
      {...rest}
      render={(props) =>
        authState.isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect to="/" />
        )
      }
    />
  );
};

export default PrivateRoute;
