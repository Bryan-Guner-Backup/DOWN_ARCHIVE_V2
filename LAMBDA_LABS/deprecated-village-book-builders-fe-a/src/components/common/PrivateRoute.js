import React from 'react';
import { Route, Redirect } from 'react-router-dom';

function PrivateRoute({ component: Component, ...rest }) {
  return (
    <Route
      {...rest}
      render={() => {
        if (localStorage.getItem('role')) {
          return <Component />;
        } else {
          return <Redirect to="/login" />;
        }
      }}
    />
  );
}

export default PrivateRoute;
