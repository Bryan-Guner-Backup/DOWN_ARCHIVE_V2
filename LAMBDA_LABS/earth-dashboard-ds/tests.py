import unittest

class TestMethods(unittest.TestCase):
   def test_add(self):
       self.assertEqual(3, 3)

if __name__ == '__main__':
   unittest.main()
