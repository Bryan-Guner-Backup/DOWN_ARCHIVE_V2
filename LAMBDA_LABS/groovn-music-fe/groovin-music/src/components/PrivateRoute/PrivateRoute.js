import React from 'react';
import { Redirect } from 'react-router-dom';

import { gql } from 'apollo-boost';
import { useQuery } from '@apollo/react-hooks';

import HomePage from '../Home/Homepage.js';

const USER_QUERY = gql`
  query currentUser {
    currentUser {
      displayName
    }
  }
`;

function PrivateRoute() {
  const { loading, error, data } = useQuery(USER_QUERY, {
    notifyOnNetworkStatusChange: true,
  });

  if (loading) {
    return <div>Loading...</div>;
  } else if (error) {
    return <div>{error.message}</div>;
  }

  console.log(data);
  const { currentUser } = data;

  if (currentUser !== null) {
    return <HomePage />;
  } else {
    console.log('redirect to Login');
    return <Redirect to="/landing-page" />;
  }
}

export default PrivateRoute;
