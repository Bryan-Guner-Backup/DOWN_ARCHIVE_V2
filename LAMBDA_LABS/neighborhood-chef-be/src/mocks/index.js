
const mocks = {       
    Query: () => ({
        getAllUsers: () => ([{Address: "das"},{Address: "das"},{Address: "das"},])
    }),
    Int: () => 2,
    Float: () => 2.1
};


module.exports = mocks;