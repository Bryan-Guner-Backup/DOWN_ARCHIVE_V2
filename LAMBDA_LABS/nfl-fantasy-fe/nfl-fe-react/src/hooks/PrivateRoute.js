import React from "react";
import { Route, Redirect }  from "react-router-dom";

const PrivateRoute = ({ component: Component, ...rest }) => {
    return (
        <Route 
        {...rest}
        render={() => { 
            if (localStorage.getItem("fan-token")) {
                return <Component />;
            } else {
                console.log(
                    "PrivateRoute.js: Route.render: redirecting"
                );
                return <Redirect to="/login" />;
            } 
        }}
        />
    );
};

export default PrivateRoute; 