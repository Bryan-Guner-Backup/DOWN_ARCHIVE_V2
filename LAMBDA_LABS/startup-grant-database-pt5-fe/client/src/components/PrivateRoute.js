import React from 'react';

//importing route and redirect to redirect user to login if not autheticated
import { Route, Redirect } from 'react-router-dom';

//Same API as `<Route />
//  spreading in ...rest.
const PrivateRoute = ({ component: Component, ...rest }) => (
  // It renders a `<Route />` and passes all the props through to it.
  <Route
    {...rest}
    render={props =>
      // It checks if the user is authenticated, if they are,
      // it renders the "component" prop. If not, it redirects
      // the user to /login.
      localStorage.getItem('authorization') ? (
        <Component {...props} />
      ) : (
        <Redirect to="/login" />
      )
    }
  />
);

export default PrivateRoute;
