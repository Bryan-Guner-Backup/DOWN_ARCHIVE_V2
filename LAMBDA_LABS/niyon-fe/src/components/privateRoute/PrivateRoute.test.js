import React from 'react'
import toJSON from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { PrivateRoute } from './PrivateRoute'
import { Redirect } from 'react-router-dom'
import { mockDiv } from '../../tests/mocks'

const setUp = (props = {}) => {
  const expectedProps = {
    component: mockDiv,
    isAuthenticated: false,
    ...props
  }
  const wrapper = shallow(<PrivateRoute {...expectedProps} />)
  return {
    wrapper,
    expectedProps
  }
}

describe('<Routes />', () => {
  let component
  beforeEach(() => {
    component = shallow(<PrivateRoute />)
  })

  it('should pass snapshot testing', () => {
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should render the component correctly', () => {
    expect(component.exists()).toBe(true)
  })
  describe('Verifying Routes are secured', () => {
    it('should pass proper props to the routes', () => {
      const { wrapper } = setUp({ path: '/lambda-school' })
      expect(wrapper.find('Route').prop('path')).toBe('/lambda-school')
    })
    it('should redirect the user to login page if user has not been authenticated', () => {
      const { wrapper } = setUp()
      const Component = wrapper.prop('render')
      const redirect = shallow(<Component location="/lambda-school" />)
      expect(redirect.find(Redirect).props()).toEqual({ to: '/login' })
    })

    it('should render the correct component if the user has been authenticated', () => {
      const { wrapper } = setUp({ isAuthenticated: true })
      const Component = wrapper.prop('render')
      const authComponent = shallow(<Component location="/lambda-school" />)
      expect(authComponent.props()).toEqual({ location: '/lambda-school' })
    })
  })
})
