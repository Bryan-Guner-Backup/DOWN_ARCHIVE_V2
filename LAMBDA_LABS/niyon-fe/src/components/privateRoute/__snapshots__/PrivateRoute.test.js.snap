// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should pass snapshot testing 1`] = `
<Route
  render={[Function]}
/>
`;
